d2cli/watch.go

```diff
@@ -430,15 +432,23 @@ func (w *watcher) handleRoot(hw http.ResponseWriter, r *http.Request) {
 	<meta charset="UTF-8">
 	<meta name="viewport" content="width=device-width, initial-scale=1.0">
 	<title>%s</title>
-	<script src="./static/watch.js"></script>
-	<link rel="stylesheet" href="./static/watch.css">
-	<link id="favicon" rel="icon" href="./static/favicon.ico">
+	<script src="/static/watch.js"></script>
+	<link rel="stylesheet" href="/static/watch.css">
+	<link id="favicon" rel="icon" href="/static/favicon.ico">
 </head>
 <body data-d2-dev-mode=%t>
 	<div id="d2-err" style="display: none"></div>
 	<div id="d2-svg-container"></div>
 </body>
 </html>`, filepath.Base(w.outputPath), w.devMode)
+
+	// if path is "/x.svg", we just want "x"
+	split := strings.Split(strings.TrimPrefix(r.URL.Path, "/"), ".")
+	boardPath := strings.Join(split[:len(split)-1], ".")
+	if boardPath != w.boardPath {
```

Comment:
What's the purpose of this check? Shouldn't we always render the requested board path?

Reply:
prevent duplication of compile requests on the initial load

Reply:
Not clear to me how this accomplishes that. Can you elaborate? Also, fairly certain this is racey, I don't think you can modify `w.boardPath` like that without putting it under a mutex as the compilation occurs in the background.

Reply:
> Can you elaborate?

I just figure on initial load it's already compiling (behavior before this PR). So I don't want to add another recompile on top of that

> fairly certain this is racey

it calls `requestCompile()`, which dodges races with channels.

Reply:
will address in another PR if needed

Reply:
That doesn't dodge the race. You can't modify a variable that's read in another goroutine without synchronization. The channel send merely guarantees that the compileLoop goroutine will run a compile at some point in the future, not that it isn't already running one.

Reply:
i did `go install -race .`, then reran the manual test in the PR description, and it didn't say there was a race.

Reply:
The race detector never produces false positives, but it does produce false negatives. It can't prove your program doesn't have races, only that it does.

Reply:
will look into this again later https://github.com/terrastruct/d2/issues/1508